pkg/k8s/event_handlers_test.go

```diff
@@ -21,65 +19,57 @@ const (
+	a := assert.New(t)
 
-		It("Should add the event to the announcement channel", func() {
-			podAddChannel := events.Subscribe(announcements.PodAdded)
-			defer events.Unsub(podAddChannel)
-
-			pod := tests.NewPodFixture(testNamespace, "pod-name", tests.BookstoreServiceAccountName, tests.PodLabels)
-			eventTypes := EventTypes{
-				Add:    announcements.PodAdded,
-				Update: announcements.PodUpdated,
-				Delete: announcements.PodDeleted,
-			}
-			handlers := GetKubernetesEventHandlers(testInformer, testProvider, shouldObserve, eventTypes)
-			handlers.AddFunc(&pod)
-			an := <-podAddChannel
-			Expect(len(podAddChannel)).To(Equal(0))
-
-			// Pubsub msg
-			pubsubMsg, castOk := an.(events.PubSubMessage)
-			Expect(castOk).To(BeTrue())
-			Expect(pubsubMsg.AnnouncementType).To(Equal(announcements.PodAdded))
-			Expect(pubsubMsg.OldObj).To(BeNil())
-
-			// Cast New obj, expect v1.Pod
-			podObj, castOk := pubsubMsg.NewObj.(*corev1.Pod)
-			Expect(castOk).To(BeTrue())
-			Expect(podObj.Name).To(Equal("pod-name"))
-			Expect(podObj.Namespace).To(Equal(testNamespace))
-		})
-
-		It("Should not add the event to the announcement channel", func() {
-			podAddChannel := events.Subscribe(announcements.PodAdded)
-			defer events.Unsub(podAddChannel)
-
-			var pod corev1.Pod
-			pod.Namespace = "not-a-monitored-namespace"
-			handlers := GetKubernetesEventHandlers(testInformer, testProvider, shouldObserve, EventTypes{})
-			handlers.AddFunc(&pod)
-			Expect(len(podAddChannel)).To(Equal(0))
-		})
-	})
-
-	Context("create getNamespace", func() {
-		It("gets the namespace name", func() {
-			ns := uuid.New().String()
-			pod := tests.NewPodFixture(ns, uuid.New().String(), tests.BookstoreServiceAccountName, tests.PodLabels)
-			actual := getNamespace(&pod)
-			Expect(actual).To(Equal(ns))
-		})
-	})
-})
+	shouldObserve := func(obj interface{}) bool {
+		object, ok := obj.(metav1.Object)
+		if !ok {
+			return false
+		}
+		return testNamespace == object.GetNamespace()
+	}
+
+	podAddChannel := events.Subscribe(announcements.PodAdded)
+	defer events.Unsub(podAddChannel)
+
+	pod := tests.NewPodFixture(testNamespace, "pod-name", tests.BookstoreServiceAccountName, tests.PodLabels)
+	eventTypes := EventTypes{
+		Add:    announcements.PodAdded,
+		Update: announcements.PodUpdated,
+		Delete: announcements.PodDeleted,
+	}
+	handlers := GetKubernetesEventHandlers(testInformer, testProvider, shouldObserve, eventTypes)
+	handlers.AddFunc(&pod)
+	an := <-podAddChannel
+	a.Len(podAddChannel, 0)
+
+	// Pubsub msg
+	pubsubMsg, castOk := an.(events.PubSubMessage)
+	a.True(castOk)
+	a.Equal(pubsubMsg.AnnouncementType, announcements.PodAdded)
+	a.Nil(pubsubMsg.OldObj)
+
+	// Cast New obj, expect v1.Pod
+	podObj, castOk := pubsubMsg.NewObj.(*corev1.Pod)
+	a.True(castOk)
+	a.Equal(podObj.Name, "pod-name")
+	a.Equal(podObj.Namespace, testNamespace)
+
+	podAddChannel = events.Subscribe(announcements.PodAdded)
+	defer events.Unsub(podAddChannel)
```

Comment:
I can't imagine this is actually a problem, but merging the two `It`s into one function here means for `podAddChannel` the test was doing

```
Subscribe()
Unsub()
Subscribe()
Unsub()
```

where now it does

```
Subscribe()
Subscribe()
Unsub()
Unsub()
```

nbd if the test reliably passes, just an observation.

Reply:
Let's not change the behavior of the existing test unless there's a bug.

Reply:
This test is primarily checking if correct events are being subscribed to and the assert statements are helping in verifying the same. The order of Subscribe() and Unsub() has changed but that doesn't impact behavior. We should be okay here